cfn-resources/teams/cmd/resource/team-user/team_user_test.go

```diff
@@ -0,0 +1,259 @@
+	mockAtlasV2Client.EXPECT().GetUserByUsername(context.Background(), validuser1).Return(&atlasv2.CloudAppUser{Id: &validuser1}, nil, nil)
+	mockAtlasV2Client.EXPECT().GetUserByUsername(context.Background(), validuser2).Return(&atlasv2.CloudAppUser{Id: &validuser2}, nil, nil)
+
+	// Call the FilterOnlyValidUsernames function
+	validUsers, _, err := FilterOnlyValidUsernames(mockAtlasV2Client, usernames)
+
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	// Check that the returned slice contains only the valid usernames
+	assert.Equal(t, len(validUsers), 2)
+	assert.Equal(t, validuser1, *validUsers[0].Id)
+	assert.Equal(t, validuser2, *validUsers[1].Id)
+}
+
+func TestFilterOnlyValidUsernamesWithInvalidInput(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockAtlasV2Client := mock.NewMockTeamUsersAPI(mockCtrl)
+
+	// Create a slice of usernames, including some valid and some invalid usernames
+	validuser1 := "validuser1"
+	invaliduser1 := "invaliduser1"
+	usernames := []string{validuser1, invaliduser1}
+
+	// Set up mock expectations for successful and unsuccessful calls to GetUserByUsername
+	mockAtlasV2Client.EXPECT().GetUserByUsername(context.Background(), validuser1).Return(&atlasv2.CloudAppUser{Id: &validuser1}, nil, nil)
+	mockAtlasV2Client.EXPECT().GetUserByUsername(context.Background(), invaliduser1).Return(nil, nil, errors.New("invalid username"))
+
+	// Call the FilterOnlyValidUsernames function
+	_, _, err := FilterOnlyValidUsernames(mockAtlasV2Client, usernames)
+
+	assert.NotNil(t, err)
+}
+
+func TestGetUsersToAddAndRemove(t *testing.T) {
+	user1 := "user1"
+	user2 := "user2"
+	user3 := "user3"
+
+	// Test cases
+	testCases := []struct {
+		currentUsers     []atlasv2.CloudAppUser
+		newUsers         []atlasv2.CloudAppUser
+		expectedToAdd    []string
+		expectedToDelete []string
+	}{
+		{
+			currentUsers: []atlasv2.CloudAppUser{
+				{Id: &user1},
+				{Id: &user2},
+			},
+			newUsers: []atlasv2.CloudAppUser{
+				{Id: &user1},
+				{Id: &user3},
+			},
+			expectedToAdd:    []string{user3},
+			expectedToDelete: []string{user2},
+		},
+		{
+			currentUsers: []atlasv2.CloudAppUser{},
+			newUsers: []atlasv2.CloudAppUser{
+				{Id: &user1},
+				{Id: &user2},
+			},
+			expectedToAdd:    []string{user1, user2},
+			expectedToDelete: []string{},
+		},
+		{
+			currentUsers: []atlasv2.CloudAppUser{
+				{Id: &user1},
+				{Id: &user2},
+			},
+			newUsers:         []atlasv2.CloudAppUser{},
+			expectedToAdd:    []string{},
+			expectedToDelete: []string{user1, user2},
+		},
+	}
+
+	// Run test cases
+	for _, testCase := range testCases {
+		toAdd, toDelete, err := GetUsersToAddAndRemove(testCase.currentUsers, testCase.newUsers)
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+		}
+
+		assert.Equal(t, testCase.expectedToAdd, toAdd)
+		assert.Equal(t, testCase.expectedToDelete, toDelete)
```

Comment:
please wrap your test case in t.Run
```suggestion
		t.Run(<test case name>, func(t *testing.T) {
			toAdd, toDelete, err := GetUsersToAddAndRemove(testCase.currentUsers, testCase.newUsers)
			require.NoError(err)
			assert.Equal(t, testCase.expectedToAdd, toAdd)
			assert.Equal(t, testCase.expectedToDelete, toDelete)
		}
```
if the test is thread safe consider also adding `t.Parallel()` inside the Run method